includes/download.clip.inc

```diff
@@ -16,7 +16,12 @@ function islandora_openseadragon_download_clip(AbstractObject $object) {
     module_load_include('inc', 'islandora_openseadragon', 'includes/utilities');
     $clip_parts = islandora_openseadragon_construct_clip_url($_GET['clip'], TRUE);
     if ($clip_parts) {
-      $filename = $object->label;
+      $filename = strtolower($object->label);
+      $map = array(
+        '/\s/' => '_',
+        "/[^0-9A-Za-z_]/" => '',
+      );
+      $filename = trim(preg_replace(array_keys($map), array_values($map), $filename));
```

Comment:
Just a tiny little little thing, but i could be paranoid. After line 24, could you check if $filename still has a value or is even valid as a file? Worst case scenario you could end with a single "_" right or just __ ? In that case we could default to some fixed name. Not a blocker really, but still good to be super safe

Reply:
@DiegoPino would a regex check suffice?
if (preg_match('/[a-zA-Z0-9]/', $filename) )
  dpm("do the work with $filename");
else
  $filename = download_clip (maybe with a date?)

Reply:
Not sure. probably yes..it seems ok

Your sanitizer all that it does is replacing spaces /tabs/etc with a '_' right? and removing invalids by replacing them with nothing, right?
You could also make it this way.. (again, no hard opinion)

```PHP
if (strlen($filename) > 0 && (strlen($filename) > substr_count( $filename , '_'))
{
dpm("do the work with $filename");
}
else {
dpm('use IAMIKenos constant name');
}
```

Ok. Now i will tell you something i just discovered of your Pull, but don't be mad. If the title of the object is in Chinese, Japanese, Korean, Hindi, Arabic, Hebrew, all this will basically give you always an empty string. But... all those languages could yield in valid File names for people running systems native in those languages and even if not, for any UTF aware OS (OSX, linux, i know nothing of windows but who knows) etc without your pull. I mean they will be UTF-8 encoded and saved. So... is that the desired outcome of this? maybe.. its not about stripping and more about encoding correctly? I mean you guys have repos in other languages right? Do those generate invalid file names? Sorry for keeping this open, but hey, world is diverse and quite large!

Reply:
'oly rabbit hole batman! 
That's a good point you bring up I suppose. I faltered down this course with [RFC2616](https://tools.ietf.org/html/rfc2616) in consideration.
You are correct in that, my the aim of my pull request is to strip non-alphanumeric characters and replacing spaces with underscores. 
Its in an attempt to say, apply a standard to the filename - lowercase letters, underscores for spaces. To keep it simple, we can strip most used punctuations with a regex such as */'\",\.\?\:\;\(\)\&\!/*

Reply:
@DiegoPino hope we're getting there 😄

Reply:
@IAMlKeno i like this! Do you need some internal feedback on DGI first? Or are we good?

Reply:
I'm sending this through the ropes internally.